### PR TITLE
Issue #3976: Inputs by Test split and organised for EmptyForIteratorPad

### LINKED ISSUE #3976: Split and Organize Checkstyle inputs by Test for EmptyForIteratorPad
subtask of #3891 .

do a copy of all test input files to separate folder for EmptyForIteratorPad .
Not any other Check should use new inputs.

### PR DESCRIPTION
**Pull request by shivanshsoni**
fixes #3976 .
It was assigned to me and discussion link:  https://groups.google.com/forum/#!topic/checkstyle-devel/ee95NUc09p0

### CODE REVIEW

**Comment by MEZk** [comment]
@shivanshsoni 
Commit message has invalid format.
Please read https://github.com/checkstyle/checkstyle/wiki/To-student-of-GSoC-2017 and https://github.com/checkstyle/checkstyle/wiki/PR-rules .

An example of correct commit message: 7e8aff1

**Comment by shivanshsoni** [comment]
Sir why my changes faliling checks?

**Comment by shivanshsoni** [comment]
please suggest some changes and please give your review on it @romani

**Comment by shivanshsoni** [comment]
okk sir I will squash all commits into one

**Comment by shivanshsoni** [comment]
But please tell me why I am failing these checks?

**Comment by rnveach** [comment]
@shivanshsoni Reading travis will tell you your issues.
https://travis-ci.org/checkstyle/checkstyle/jobs/209993811#L886-L898

1) you changed location where tests look for inputs, but you didn't move inputs to that location.
2) Like @MEZk said, commit message must be changed to our required pattern.

**Comment by shivanshsoni** [comment]
I am closing this pull request and making a new pull request

**Comment by rnveach** [comment]
@shivanshsoni You don't need to keep closing/opening PR. If you made changes, push them to the branch and PR will automatically update.

**Comment by shivanshsoni** [comment]
@rnveach Okk Sir

**Comment by Vladlis** [comment]
@shivanshsoni 
https://github.com/checkstyle/checkstyle/wiki/PR-rules

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/EmptyLineSeparatorCheck.java [hunk 1]
```diff
@@ -218,13 +218,13 @@
             "empty.line.separator.multiple.lines.inside";
 
     /** Allows no empty line between fields. */
-    private boolean allowNoEmptyLineBetweenFields;
+    private boolean allowNoEmptyLineBetweenFields = true;
```
Comment (rnveach) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/EmptyLineSeparatorCheck.java]: We are not expecting any changes under the `main` folder.

File: src/test/java/com/puppycrawl/tools/checkstyle/checks/whitespace/EmptyForIteratorPadCheckTest.java [hunk 2]
```diff
@@ -49,7 +49,9 @@ public void setUp() {
     @Override
     protected String getPath(String filename) throws IOException {
         return super.getPath("checks" + File.separator
-                + "whitespace" + File.separator + filename);
+                + "whitespace" + File.separator 
+                + "IteratorPad" + File.separator
```
Comment (rnveach) [inline src/test/java/com/puppycrawl/tools/checkstyle/checks/whitespace/EmptyForIteratorPadCheckTest.java]: lowercase for package names.

Comment (rnveach) [inline src/test/java/com/puppycrawl/tools/checkstyle/checks/whitespace/EmptyForIteratorPadCheckTest.java]: folder should name should be `emptyforiteratorpad` after test name.

File: src/test/java/com/puppycrawl/tools/checkstyle/checks/whitespace/EmptyForIteratorPadCheckTest.java [hunk 3]
```diff
@@ -66,7 +68,7 @@ public void testDefault() throws Exception {
             "43:32: " + getCheckMessage(MSG_WS_FOLLOWED, ";"),
             "55:11: " + getCheckMessage(MSG_WS_FOLLOWED, ";"),
         };
-        verify(checkConfig, getPath("InputForWhitespace.java"), expected);
+        verify(checkConfig, getPath("InputForIteratorPad.java"), expected);
```
Comment (rnveach) [inline src/test/java/com/puppycrawl/tools/checkstyle/checks/whitespace/EmptyForIteratorPadCheckTest.java]: inputs should be named after test. `InputEmptyForIteratorPad...`

**Comment by rnveach** [comment]
@shivanshsoni squash all commits and fix all broken CIs.

**Comment by shivanshsoni** [comment]
folder should name should be emptyforiteratorpad after test name.   What does this mean?@rnveach

**Comment by shivanshsoni** [comment]
which folder?

**Comment by rnveach** [comment]
@shivanshsoni As you are fixing CI, please squash all commits to 1, keep them at 1, and rebase off the latest master.

> folder should name should be emptyforiteratorpad after test name. What does this mean?

https://github.com/checkstyle/checkstyle/pull/3978#discussion_r105749082
Look at line where comment was made. You set `getPath` to be in `IteratorPad`, but the folder name should be named after the test, so it should be `emptyforiteratorpad`.

**Pull request closed without merging**